arch/x86/core/ia32/swap.S

```diff
@@ -65,30 +65,7 @@
  * unsigned int arch_swap (unsigned int eflags);
  */
 
-.macro read_tsc var_name
-	push %eax
-	push %edx
-	rdtsc
-	mov %eax,\var_name
-	mov %edx,\var_name+4
-	pop %edx
-	pop %eax
-.endm
 SECTION_FUNC(TEXT, arch_swap)
-#ifdef CONFIG_EXECUTION_BENCHMARKING
-	/* Save the eax and edx registers before reading the time stamp
-	* once done pop the values.
-	*/
-	push %eax
-	push %edx
-	rdtsc
-	mov %eax,arch_timing_swap_start
-	mov %edx,arch_timing_swap_start+4
-	pop %edx
-	pop %eax
-#endif
-
-
 #if defined(CONFIG_TRACING)
 	pushl	%eax
```

Comment:
I think the push/pop of EAX can be dropped here, sys_trace_thread_switched_out takes no arguments and we're not using it in this ASM code at the time it is called.